src/adapters/models/llama/adapter_model.py

```diff
@@ -149,3 +152,14 @@ def prepare_inputs_for_generation(
             }
         )
         return model_inputs
+
+    def _load_pretrained_model(cls, model, state_dict, loaded_keys, *args, **kwargs):
```

Comment:
Why is this rewriting of the state dict required for us but not for HF?

Reply:
I believe this is the case because when using Hugging Face, the question answering llama is loaded using the `AutoModelForQuestionAnswering` class, which utilises the models in `MODEL_FOR_QUESTION_ANSWERING_MAPPING_NAMES` and this contains `LlamaForQuestionAnswering`. `LlamaForQuestionAnswering` has "transformer" as the base model (https://github.com/huggingface/transformers/blob/2209b7afa04b3a6366350065f541e9248d6663c2/src/transformers/models/llama/modeling_llama.py#L1460)

But we load all models using the same `AutoAdapterModel` class: In `LlamaAdapterModel`, we set `self.model = LlamaModel(config)` and because of this the `state_dict` and `model` keys must have "model" as the base model.